client/notification.js

```diff
@@ -18,25 +18,74 @@ function NotifyClient() {
 
 /**
  *
+ * @param {String} type
  * @param {String} templateId
  * @param {String} to
  * @param {Object} personalisation
+ * @param {String} reference
  *
  * @returns {Object}
  */
-function createNotificationPayload(templateId, to, personalisation) {
+function createNotificationPayload(type, templateId, to, personalisation, reference) {
+
   var payload = {
-    template: templateId,
-    to: to
+    template_id: templateId
   };
 
+  if (type == 'email') {
+    payload.email_address = to
+  } else if (type == 'sms') {
+    payload.phone_number = to
```

Comment:
v minor quibble but it would be good to be consistent on semicolons at end of lines